src/utils/configureGlobalAgent.ts

```diff
@@ -0,0 +1,18 @@
+/*---------------------------------------------------------------------------------------------
+*  Copyright (c) Microsoft Corporation. All rights reserved.
+*  Licensed under the MIT License. See License.txt in the project root for license information.
+*--------------------------------------------------------------------------------------------*/
+
+import { bootstrap } from "global-agent";
+import { workspace } from "vscode";
+
+export async function configureGlobalAgent(): Promise<void> {
+
+    workspace.getConfiguration('http.proxy')
+
+    // lowercase environment variables take precedence over uppercase ones
+    process.env.GLOBAL_AGENT_HTTPS_PROXY = workspace.getConfiguration('http').get('proxy') || process.env.https_proxy || process.env.HTTPS_PROXY || process.env.http_proxy || process.env.HTTP_PROXY;
```

Comment:
Is this a standard order or is it something we should try to document somewhere?

Reply:
It's not standard, I followed the order VS Code used to configure their proxy.

Reply:
Good idea, a link to the source or doc or whatever indicating that order is helpful.